src/cryptography/x509.py

```diff
@@ -1344,3 +1344,52 @@ def extensions(self):
         """
         Returns an Extensions object containing a list of Revoked extensions.
         """
+
+
+class CertificateSigningRequestBuilder(object):
+    def __init__(self, version=Version.v1, subject_name=None, extensions=[]):
+        """
+        Creates an empty X.509 certificate request (v1).
+        """
+        self._version = Version.v1
+        self._subject_name = subject_name
+        self._extensions = extensions[:]
+
+    def set_version(self, version):
```

Comment:
OK.  I'll send in a separate PR to fix the CSR test vectors.  I'll also remove the `set_version()` function since it seems like it can only be misused at the moment.